Add support for values

ChristopherRogers1991 opened the pull request:
## Description
Extends IoTRequest to allow for integer values to be included in the requests. Includes the introduction of version info, allowing us to extend IoTRequest without requiring all skills to update immediately.

## How to test
Use the latest version of the iot-control-skill and the homeassistant@feature/commonIoT skill to set heat/temperature/ac to a specific value.

## Contributor license agreement signed?
CLA
 - [x] (Whether you have signed a [CLA - Contributor Licensing Agreement](https://mycroft.ai/cla/)

forslund commented on mycroft/skills/common_iot_skill.py:
```diff
@@ -305,6 +355,16 @@ def register_entities_and_scenes(self):
+
+        By default, this returns IoTRequestVersion.V1. Subclasses
+        should override this to indicate higher levels of support.
+        """
```
Should the supported features for each version be described here (or perhaps up in the version enum) so it's included in the readthedocs documentation?

forslund commented on mycroft/skills/common_iot_skill.py:
```diff
@@ -155,30 +191,40 @@ def __init__(self,
             entity='"{}"'.format(self.entity) if self.entity else None,
-            scene='"{}"'.format(self.scene) if self.scene else None
+            scene='"{}"'.format(self.scene) if self.scene else None,
+            value='"{}"'.format(self.value) if self.value else None
```
Just realized another thing... if value is 0 this will fail and set the value to None. Or is that intended?

ChristopherRogers1991 commented:
@forslund, thanks for the review. I've added some additional documentation, and fixed the `__repr__` method to account for `0` (excellent catch on this, by the way). Let me know if you have any other thoughts or feedback.

forslund commented:
Looks good now, it manges to set my house temperature to a nice and cozy 0 degrees now :)

Merging
